File: src/main/java/com/example/authapp/dto/response/AuthResponse.java
package com.example.authapp.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {
    private String accessToken;
    private String refreshToken;
    @Builder.Default
    private String tokenType = "Bearer";
    private String name;
    private String email;
}

File: flutter_app/lib/models/auth_response.dart
class AuthResponse {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final String name;
  final String email;

  AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.name,
    required this.email,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'] ?? '',
      refreshToken: json['refreshToken'] ?? '',
      tokenType: json['tokenType'] ?? 'Bearer',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
    );
  }
}
